master/buildbot/worker/openstack.py

```diff
@@ -231,39 +234,58 @@ def _getImage(self, build):
         # function should return the image's UUID to use.
         image = self.image
         if callable(image):
-            image_uuid = image(self.novaclient.images.list())
+            # novaclient has renamed images to glance
+            if hasattr(self.novaclient, "glance"):
+                image_uuid = image(self.novaclient.glance.list())
+            else:
+                image_uuid = image(self.novaclient.images.list())
         else:
             image_uuid = yield build.render(image)
         return image_uuid
 
+    @defer.inlineCallbacks
+    def _getFlavor(self, build):
+        # If flavor is callable, then pass it the list of flavors. The
+        # function should return the flavor's UUID to use.
+        flavor = self.flavor
+        if callable(flavor):
```

Comment:
I know I used `callable` for the `images` method but I think only using a renderer fits better with the rest of Buildbot.

Reply:
it's actually quite useful, because otherwise you cannot easily resolve `m1.medium` into UUID.

Reply:
Example code:
```.py
# Used by openstack client - return matching flavor uuid
def os_get_flavor(flavors):
    want_flavor = util.Interpolate('%(prop:os_flavos)s')
    for flavor in flavors:
        if flavor.name == want_flavor:
            return flavor.id
    return want_flavor

# Used by openstack client - return matching image uuid
def os_get_image(images):
    want_image = util.Interpolate('%(prop:os_image)s')
    for image in images:
        if image.name == want_image:
            return image.id
    return want_image
```

Reply:
Actually, maybe we could just move this code inside the worker? Would that be ok?

Reply:
I would definitely prefer to be able to use the name instead of the UUID so I think it belongs in Buildbot so others can also benefit.